### ISSUE TITLE
Unable to use helpers/filters in computed data (Cannot find module '.eleventy')

### ISSUE THREAD

**Issue by patrulea** [11ty/eleventy#3134]
### Operating system

macOS Sonoma 14.2.1

### Eleventy

2.0.1

### Describe the bug

I’m trying to use a filter to compute front-matter data but I can’t import the Eleventy configuration file to reference the filters I defined there.
```js
// /layouts/base.webc
{
  eleventyComputed: {
    favicon: data => {
      const sImage = require("../.eleventy.js").getFilter("sImage");
      return sImage(data.general.meta.favicon);
    }
}
```
```bash
Problem writing Eleventy templates: (more in DEBUG output)
Cannot find module '../.eleventy.js'
Require stack:
- [root]/node_modules/gray-matter/lib/engines.js
- [root]/node_modules/gray-matter/lib/defaults.js
- [root]/node_modules/gray-matter/index.js
- [root]/node_modules/@11ty/eleventy/src/TemplateContent.js
- [root]/node_modules/@11ty/eleventy/src/Template.js
- [root]/node_modules/@11ty/eleventy/src/TemplateWriter.js
- [root]/node_modules/@11ty/eleventy/src/Eleventy.js
- [root]/node_modules/@11ty/eleventy/cmd.js (via Error)
```

**Comment by patrulea** [comment]
This is the filter in particular, if it helps. It takes an image reference object from a Sanity CMS document and outputs its URL.
```js
// .eleventy.js
const builder = require("@sanity/image-url");
module.exports = eleventyConfig => {
  eleventyConfig.addFilter("sImage", (source) => builder({
    projectId: process.env.SANITY_STUDIO_PROJECT_ID,
    dataset: process.env.SANITY_STUDIO_DATASET,
    apiVersion: process.env.SANITY_STUDIO_API_VERSION,
    token: process.env.SANITY_STUDIO_API_TOKEN,
    useCdn: true
  }).image(source).url());
}
```
This filter works well within the template, outside the front-matter “fence” (`---`).

**Comment by uncenter** [comment]
Yeah it's not the filter, the issue is about `require()`-ing the Eleventy config.

**Comment by Snapstromegon** [comment]
I think this one should be solved by #3130, you could then do
```js
data => {
  const sImage = data.eleventy.config.getFilter("sImage");
  return sImage(data.general.meta.favicon);
}
```

As of right now you could put your filter into a separate file (e.g. _filters/sImage.js_) and import that from _.eleventy.js_ and your data.

**Comment by uncenter** [comment]
#3130 really is coming in handy!

**Comment by patrulea** [comment]
> I think this one should be solved by #3130.
Yes! that’s exactly what I needed. I’ll keep an eye on the updates as that would really come in handy for this project I’m working on.
